### DISCUSSION TITLE
Help with setting up an RPC provider to sign transactions

### DISCUSSION THREAD

**Discussion by rogerjbos** [ProjectOpenSea/opensea-js#1189] [General]
I have some OpenSea SDK code working to fetch data using the default infura provider, but now I would like to add a provider that I can use to sign transactions, but I can't figure it out based on Googling.  Here is what I have tried:
```
import HDWalletProvider from '@truffle/hdwallet-provider';
const provider = new HDWalletProvider(WALLET_MNEMONIC, ALCHEMY_API_URL);
console.log("provider is ", provider)
const openseaSDK = new OpenSeaSDK(provider, {
  chain: Chain.Mainnet,
  apiKey: OPENSEA_API_KEY,
});

```
   throw new Error("Either a provider or custom signer with provider must be provided");

Could anyone please provide any additional hints on how to configure a RPC provider based on my mnemonic (or private keys)?

**Comment by rogerjbos** [comment]
Reading the ethers 5.7 documentation, I am now using the code below.  I think I am really close.  I have a wallet with a private key and to test it out I am trying to place an offer, but I am getting an error 
`Error: could not detect network (event="noNetwork", code=NETWORK_ERROR, version=providers/5.7.2)`

Any suggestions would be appreciated: 
```
import { ethers } from "ethers";
import { OpenSeaSDK, Chain } from "opensea-js";
import { INFURA_URL, OPENSEA_API_KEY, WALLET_ADDRESS, PRIVATE_KEY } from './constants.js';
// import { getCollectionFloorPrice } from './nftFunctions.js'

let n = ethers.providers.getNetwork("mainnet");
console.log("network is ", n);

const provider = new ethers.providers.JsonRpcProvider(INFURA_URL, ethers.providers.getNetwork("mainnet"));
const walletWithProvider = new ethers.Wallet(PRIVATE_KEY, provider);

const openseaSDK = new OpenSeaSDK(walletWithProvider, {
  chain: Chain.Mainnet,
  apiKey: OPENSEA_API_KEY,
});

const my_asset = {
  tokenAddress: '0x4135063DC85190660ed08790f59Bc711d8b404c0', // newhere
  tokenId: '3333',
  tokenStandard: 'ERC721'
};

const offerAmount = .00999;
const offer = await openseaSDK.createBuyOrder({
  asset: my_asset,
  accountAddress: WALLET_ADDRESS,
  // Value of the offer, in units of the payment token (or wrapped ETH if none is specified):
  startAmount: offerAmount, 
});
console.log(offer);
```

**Reply by ryanio** [reply]
nice figuring it out, i will try to add the ethers.Wallet init to our docs tomorrow for others to more easily find.

i think you also need to pass the wallet as the 4th constructor arg, so change to 
```
const openseaSDK = new OpenSeaSDK(walletWithProvider, {
  chain: Chain.Mainnet,
  apiKey: OPENSEA_API_KEY,
}, undefined, walletWithProvider );
```
this is something on my list to fix in next version

**Comment by rogerjbos** [comment]
Thank you for the response.  I was still getting the same error, so that prompted me to look at other things, and I tried using Alchemy instead of Infura and that worked.  So there must be something wrong with my Infura API config.  

So just to confirm, your suggestion works:
```const openseaSDK = new OpenSeaSDK(walletWithProvider, {
  chain: Chain.Mainnet,
  apiKey: OPENSEA_API_KEY,
}, undefined, walletWithProvider );
```
But the simpler one also works:
```
const openseaSDK = new OpenSeaSDK(walletWithProvider, {
  chain: Chain.Mainnet,
  apiKey: OPENSEA_API_KEY,
});
```

Thanks again!
